frontend/src/components/utils/autoRefreshHandler.tsx

```diff
@@ -49,48 +50,64 @@ export default (
           items: Array<any>(),
           itemsLength: config.count,
           display: false,
-          hasMore: true
+          hasMore: true,
+          loading: false
         });
       }
     }
 
+    regularFetchInfo = () => {
+      if (config.dashboard) {
+        this.setState({ hasMore: false });
+      }
+      if (this.state.itemsLength > 0) {
+        this.fetchInfo(this.state.itemsLength);
+        if (this.timer !== null) {
+          this.timer = setTimeout(this.regularFetchInfo, 10000);
+        }
+      }
+    };
+
     fetchInfo = (count: number) => {
       config
         .fetchDataFn(count)
         .then((items: any) => {
           if (items.length > 0) {
-            this.setState({ items, display: true });
+            this.setState({ items, itemsLength: items.length, display: true });
+            if (items.length < config.count) {
+              this.setState({ hasMore: false });
+            }
```

Comment:
It is still possible that we will trigger 2 renders instead of 1 here. Just as a pattern, collect the state changes into an object and the apply that once:

```js
let newState;
if ... {
   newState = ...
   if ... {
       newState.hasMore = false;
   }
} else { ... }
this.setState(newState)
```